pkg/raft/raft.go

```diff
@@ -2076,8 +2068,6 @@ func stepCandidate(r *raft, m pb.Message) error {
 			// m.Term > r.Term; reuse r.Term
 			r.becomeFollower(r.Term, r.lead)
 		}
-	case pb.MsgTimeoutNow:
```

Comment:
It is removed because `MsgTimeoutNow` is caught by `IsMsgFromLeader`. With this change, a candidate will now handle the `MsgTimeoutNow` as an instruction to force campaign at the next term, instead of ignoring it.

Reply:
Yes, this is new behavior. But it's desirable behavior. I've extracted it into https://github.com/cockroachdb/cockroach/pull/133490.

> Also, do we need to assume MsgTimeoutNow came from the leader? It's a forceful campaign, and in theory any node could randomly decide to do that without the leader's involvement. Perhaps the uncertainty comes from the fact that it's used both for forced campaigns and leader transfers (and in the latter case it makes sense to say that the message is made on behalf of this leader).

It used to be disruptive but safe to allow any node to decide to perform a force campaign. With leader fortification and leader leases, it is now very unsafe. A force campaign will instruct any voter to ignore the fortification promise that it has made to the leader, which could cause a lease expiration regression (overlapping leases). Force campaigns are only safe if the leader is the one initiating them, because it will step down to a follower at the same time.

> used both for forced campaigns and leader transfers

The notion of a "forced campaign", separate from a leader transfer is relatively new. It came from https://github.com/cockroachdb/cockroach/pull/104969 and I'd like to eliminate the concept with something like https://github.com/cockroachdb/cockroach/pull/133308.